chroma_core/models/ntp.py

```diff
@@ -56,14 +57,34 @@ class StopChronyStep(Step):
     idempotent = True
 
     def run(self, kwargs):
-        return self.invoke_rust_agent_expect_result(kwargs["fqdn"], "stop_unit", "chronyd.service")
+        from chroma_core.services.job_scheduler.agent_rpc import AgentException
+
+        try:
+            return self.invoke_rust_agent_expect_result(kwargs["fqdn"], "stop_unit", "chronyd.service")
+        except AgentException as e:
+            t, v, tb = sys.exc_info()
+
+            if "Unknown busctl" in str(e):
```

Comment:
Is there a more clear way to say no such unit? 🙂

Reply:
Yeah, I agree this is frustrating and ugly. If you know of a better way to check please let me know.

Reply:
Instead of trying to stop and handling the error. We should just check the state first and then stop iff it's already installed / running